### PR TITLE
Make NVCC wrapper identify compiler features

### PR DESCRIPTION
**Pull request by jjwilke**

### CODE REVIEW

**Comment by dalg24** [comment]
Add to whitelist

File: bin/nvcc_wrapper [hunk 1]
```diff
@@ -191,7 +191,30 @@ do
     cuda_args="$cuda_args $1"
     ;;
   #Handle c++11
-  --std=c++11|-std=c++11|--std=c++14|-std=c++14|--std=c++1y|-std=c++1y|--std=c++17|-std=c++17|--std=c++1z|-std=c++1z)
+  --std=c++1y|-std=c++1y|--std=c++1z|-std=c++1z|--std=gnu++1y|-std=gnu++1y|--std=gnu++1z|-std=gnu++1z)
+    if [ $stdcxx_applied -eq 1 ]; then
+       echo "nvcc_wrapper - *warning* you have set multiple optimization flags (-std=c++1* or --std=c++1*), only the first is used because nvcc can only accept a single std setting"
+    else
+       if [ ! -z "${KOKKOS_CMAKE_CXX_STANDARD}" ]; then
```
[the remark this answers is not in the document]
Comment (crtrott) [reply bin/nvcc_wrapper]: You need to add this change also at https://github.com/kokkos/nvcc_wrapper

File: cmake/kokkos_compiler_id.cmake [hunk 2]
```diff
@@ -55,6 +55,7 @@ ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL NVIDIA)
   IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 7.0.28)
     MESSAGE(FATAL_ERROR "${KOKKOS_MESSAGE_TEXT}")
   ENDIF()
+  SET(CMAKE_CXX_EXTENSIONS OFF CACHE BOOL "Kokkos turns off CXX extensions" FORCE)
```
Comment (dalg24) [inline cmake/kokkos_compiler_id.cmake]: This feels so wrong.

**Comment by crtrott** [comment]
This now passed on Apollo too:
```
Running on machine: apollo
Repository Status:  0fa8ceaf084dac655ae21e4e05b6fccf3e14fb84 Fix another shadow warning


sems-cmake/3.5.2(3):ERROR:150: Module 'sems-cmake/3.5.2' conflicts with the currently loaded module(s) 'sems-cmake/3.12.2'
sems-cmake/3.5.2(3):ERROR:102: Tcl command execution failed: conflict sems-cmake

Going to test compilers:  gcc/4.8.4 gcc/5.3.0 intel/16.0.1 clang/3.9.0 clang/6.0 cuda/9.1 hpx/1.2.1
Testing compiler gcc/4.8.4
  Starting job gcc-4.8.4-OpenMP-release
  PASSED gcc-4.8.4-OpenMP-release
Testing compiler gcc/5.3.0
  Starting job gcc-4.8.4-Pthread-release
  PASSED gcc-4.8.4-Pthread-release
Testing compiler intel/16.0.1
  Starting job gcc-5.3.0-Serial-release
  PASSED gcc-5.3.0-Serial-release
Testing compiler clang/3.9.0
  Starting job intel-16.0.1-OpenMP-release
  PASSED intel-16.0.1-OpenMP-release
Testing compiler clang/6.0
  Starting job clang-3.9.0-Pthread_Serial-release
  PASSED clang-3.9.0-Pthread_Serial-release
  Starting job clang-6.0-Cuda_Pthread-release
  PASSED clang-6.0-Cuda_Pthread-release
Testing compiler cuda/9.1
  Starting job clang-6.0-OpenMP-release
  PASSED clang-6.0-OpenMP-release
Testing compiler hpx/1.2.1
  Starting job cuda-9.1-Cuda_OpenMP-release
  PASSED cuda-9.1-Cuda_OpenMP-release
  Starting job hpx-1.2.1-HPX-release
  PASSED hpx-1.2.1-HPX-release
#######################################################
PASSED TESTS
#######################################################
clang-3.9.0-Pthread_Serial-release build_time=150 run_time=539
clang-6.0-Cuda_Pthread-release build_time=258 run_time=416
clang-6.0-OpenMP-release build_time=126 run_time=125
cuda-9.1-Cuda_OpenMP-release build_time=346 run_time=249
gcc-4.8.4-OpenMP-release build_time=122 run_time=181
gcc-4.8.4-Pthread-release build_time=114 run_time=565
gcc-5.3.0-Serial-release build_time=133 run_time=196
hpx-1.2.1-HPX-release build_time=1182 run_time=453
intel-16.0.1-OpenMP-release build_time=462 run_time=393
```

**Pull request merged**